src/app/forms/forms.component.html

```diff
@@ -0,0 +1,48 @@
+<div>
+  <a routerLink="/login">Logout</a>
+</div>
+<div class="example-portal-outlet">
+  <div class="portal-outlet">
+    <ng-container #accordionContainer cdkPortalOutlet></ng-container>
+  </div>
+  <div class="portal-outlet">
+    <ng-container #builderContainer cdkPortalOutlet></ng-container>
+  </div>
+  <div class="portal-outlet">
+    <ng-container #elementsContainer cdkPortalOutlet></ng-container>
+  </div>
+</div>
+
+<div #comp1 class="portal-outlet">
+  <app-switch item="general"></app-switch>
+  <div *ngFor="let item of dropArray">
+    <app-switch [item]="item"></app-switch>
+  </div>
+</div>
+<div [formGroup]="form" #comp2 class="portal" >
+  <h3>Builder section</h3>
+  <div cdkDropList
+       #dropList="cdkDropList"
+       [cdkDropListData]="dropArray"
+       [cdkDropListConnectedTo]="[dragList]"
+       (cdkDropListDropped)="drop($event)">
+    <div>Put your elements here:</div>
+    <div class="builder" [ngStyle]="getStylesByType('general') | async">
```

Comment:
use ngrxPush
https://ngrx.io/guide/component/push
```suggestion
    <div class="builder" [ngStyle]="getStylesByType('general') | ngrxPush">
```